File: SDKs/amazonaws.com/elasticache/2015-02-02/python/src/sdk/models/shared/reservedcachenodemessage.py
import dataclasses
from typing import Optional
from ..shared import reservedcachenodelist as shared_reservedcachenodelist


@dataclasses.dataclass
class ReservedCacheNodeMessage:
    r"""ReservedCacheNodeMessage
    Represents the output of a <code>DescribeReservedCacheNodes</code> operation.
    """
    
    marker: Optional[str] = dataclasses.field(default=None)
    reserved_cache_nodes: Optional[list[shared_reservedcachenodelist.ReservedCacheNodeList]] = dataclasses.field(default=None)
    

File: SDKs/amazonaws.com/elasticache/2015-02-02/typescript/src/sdk/models/shared/reservedcachenodemessage.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ReservedCacheNodeList } from "./reservedcachenodelist";



// ReservedCacheNodeMessage
/** 
 * Represents the output of a <code>DescribeReservedCacheNodes</code> operation.
**/
export class ReservedCacheNodeMessage extends SpeakeasyBase {
  @SpeakeasyMetadata()
  marker?: string;

  @SpeakeasyMetadata({ elemType: ReservedCacheNodeList })
  reservedCacheNodes?: ReservedCacheNodeList[];
}
